# Aufgabe 1: Datenexploration

In dieser Aufgabe werden wir den Adult Income Datensatz analysieren und erste Einblicke gewinnen.

In [ ]:
import pandas as pd
import numpy as np

# Datensatz einlesen
df = pd.read_csv('adult.csv')

✓ Datensatz erfolgreich eingelesen


## 1.1. Erste 5 Zeilen anzeigen

In [9]:
df.head()

,age,workclass,fnlwgt,education,educational-num,marital-status,occupation,relationship,race,gender,capital-gain,capital-loss,hours-per-week,native-country,income
0,25,Private,226802,11th,7,Never-married,Machine-op-inspct,Own-child,Black,Male,0,0,40,United-States,<=50K
1,38,Private,89814,HS-grad,9,Married-civ-spouse,Farming-fishing,Husband,White,Male,0,0,50,United-States,<=50K
2,28,Local-gov,336951,Assoc-acdm,12,Married-civ-spouse,Protective-serv,Husband,White,Male,0,0,40,United-States,>50K
3,44,Private,160323,Some-college,10,Married-civ-spouse,Machine-op-inspct,Husband,Black,Male,7688,0,40,United-States,>50K
4,18,?,103497,Some-college,10,Never-married,?,Own-child,White,Female,0,0,30,United-States,<=50K


## 1.2. Anzahl der Zeilen und Spalten

In [ ]:
print(f"Anzahl der Zeilen: {df.shape[0]}")
print(f"Anzahl der Spalten: {df.shape[1]}")

Anzahl der Zeilen: 48842
Anzahl der Spalten: 15

Dimension des Datensatzes: (48842, 15)


## 1.3. Spaltennamen und deren Bedeutung

In [11]:
print("Spaltennamen:")
print(df.columns.tolist())

print("\n" + "="*60)
print("Datentypen der Spalten:")
print("="*60)
print(df.dtypes)

Spaltennamen:
['age', 'workclass', 'fnlwgt', 'education', 'educational-num', 'marital-status', 'occupation', 'relationship', 'race', 'gender', 'capital-gain', 'capital-loss', 'hours-per-week', 'native-country', 'income']

Datentypen der Spalten:
age                int64
workclass            str
fnlwgt             int64
education            str
educational-num    int64
marital-status       str
occupation           str
relationship         str
race                 str
gender               str
capital-gain       int64
capital-loss       int64
hours-per-week     int64
native-country       str
income               str
dtype: object


### Interpretation der Spaltennamen

Die meisten Spaltennamen sind intuitiv verständlich:
- **age**: Alter der Person
- **workclass**: Beschäftigungsart (z.B. Private, Self-employed, Government)
- **fnlwgt**: Final weight - Gewichtungsfaktor für demografische Hochrechnungen
- **education**: Bildungsabschluss
- **educational-num**: Numerischer Wert für Bildungsniveau
- **marital-status**: Familienstand
- **occupation**: Beruf
- **relationship**: Verwandtschaftsverhältnis im Haushalt
- **race**: Ethnische Zugehörigkeit
- **gender**: Geschlecht
- **capital-gain**: Kapitalgewinne
- **capital-loss**: Kapitalverluste
- **hours-per-week**: Wochenarbeitsstunden
- **native-country**: Herkunftsland
- **income**: Einkommen (Zielvariable: <=50K oder >50K)

## 1.4. Deskriptive Statistik mit df.describe()

In [12]:
df.describe()

,age,fnlwgt,educational-num,capital-gain,capital-loss,hours-per-week
count,48842.000000,4.884200e+04,48842.000000,48842.000000,48842.000000,48842.000000
mean,38.643585,1.896641e+05,10.078089,1079.067626,87.502314,40.422382
std,13.710510,1.056040e+05,2.570973,7452.019058,403.004552,12.391444
min,17.000000,1.228500e+04,1.000000,0.000000,0.000000,1.000000
25%,28.000000,1.175505e+05,9.000000,0.000000,0.000000,40.000000
50%,37.000000,1.781445e+05,10.000000,0.000000,0.000000,40.000000
75%,48.000000,2.376420e+05,12.000000,0.000000,0.000000,45.000000
max,90.000000,1.490400e+06,16.000000,99999.000000,4356.000000,99.000000


In [13]:
# Detailliertere Analyse potenzieller Ausreißer
print("Analyse potenzieller Ausreißer:\n")

for col in df.select_dtypes(include=[np.number]).columns:
    q1 = df[col].quantile(0.25)
    q3 = df[col].quantile(0.75)
    iqr = q3 - q1
    lower_bound = q1 - 1.5 * iqr
    upper_bound = q3 + 1.5 * iqr
    
    outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)][col]
    outlier_percentage = (len(outliers) / len(df)) * 100
    
    if outlier_percentage > 0:
        print(f"{col}:")
        print(f"  - Ausreißer-Anteil: {outlier_percentage:.2f}%")
        print(f"  - Min: {df[col].min()}, Max: {df[col].max()}")
        print(f"  - IQR-Grenzen: [{lower_bound:.2f}, {upper_bound:.2f}]")
        print()

Analyse potenzieller Ausreißer:

age:
  - Ausreißer-Anteil: 0.44%
  - Min: 17, Max: 90
  - IQR-Grenzen: [-2.00, 78.00]

fnlwgt:
  - Ausreißer-Anteil: 2.97%
  - Min: 12285, Max: 1490400
  - IQR-Grenzen: [-62586.75, 417779.25]

educational-num:
  - Ausreißer-Anteil: 3.67%
  - Min: 1, Max: 16
  - IQR-Grenzen: [4.50, 16.50]

capital-gain:
  - Ausreißer-Anteil: 8.26%
  - Min: 0, Max: 99999
  - IQR-Grenzen: [0.00, 0.00]

capital-loss:
  - Ausreißer-Anteil: 4.67%
  - Min: 0, Max: 4356
  - IQR-Grenzen: [0.00, 0.00]

hours-per-week:
  - Ausreißer-Anteil: 27.63%
  - Min: 1, Max: 99
  - IQR-Grenzen: [32.50, 52.50]



## 1.5. Zusammenfassung und Interpretation

### Datensatzgröße
Der Datensatz enthält **48.842 Zeilen** und **15 Spalten**.

### Potenzielle Ausreißer (IQR-Methode)

| Variable | Ausreißer-Anteil | Bewertung |
|----------|------------------|-----------|
| **capital-gain** | 8.26% | Stark rechtsschiefe Verteilung |
| **capital-loss** | 4.67% | Stark rechtsschiefe Verteilung |
| **hours-per-week** | 27.63% | Höchster Anteil, aber realistisch (Teilzeit/Überzeit) |
| **educational-num** | 3.67% | Normale Variation |
| **fnlwgt** | 2.97% | Gewichtungsfaktor, für ML nicht relevant |
| **age** | 0.44% | Minimal, realistisch |